### ISSUE TITLE
Rule AM06 not working correctly for BigQuery (for implicit config)

### ISSUE THREAD

**Issue by daczarne** [sqlfluff/sqlfluff#5497]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When the AM06 rule is set to `group_by_and_order_by_style = implicit` and the dialect is BigQuery not all instances of `ORDER BY` need to be considered.

### Expected Behaviour

For `ORDER BY` clauses within `ARRAY` calls to be ignored.

### Observed Behaviour

SQLFluff expects for the `ORDER BY` elements in an `ARRAY` call to be implicit, but this is not supported by BigQuery.

### How to reproduce

Consider the following query

```sql
SELECT
  poi.country_code
  , poi.po_id
  , ARRAY(
      SELECT STRUCT(
          p.product_name
          , p.sku_id
          , p.created_at
        ) AS products_purchased
      FROM UNNEST(poi.products_purchased) AS p
      ORDER BY p.created_at
    ) AS products_purchased
FROM `my_project.my_dataset.purchase_orders_products` AS poi
```

If we run `.sqlfluff lint` with dialect BigQuery and rule AM06 (for testing purposes this can be the only rule applied), SQL Fluff will complain that the `ORDER BY` clause is inconsistent. But BigQuery does not allow for that `ORDER BY` to be numbers.

### Dialect

Bigquery

### Version

- SQL Fluff: 2.3.5
- Python: 3.10.1

### Configuration

```cfg
[sqlfluff]
dialect = bigquery
disable_noqa = False
encoding = utf-8
fix_even_unparsable = False
# ignore = templating, parsing
ignore_templated_areas = True
large_file_skip_byte_limit = 0
large_file_skip_char_limit = 0
max_line_length = 200
nocolor = False
output_line_length = 80
processes = 1
runaway_limit = 10
sql_file_exts = .sql
templater = jinja
verbose = 2
warn_unused_ignores = False

rules = AM06

[sqlfluff:rules:ambiguous.column_references]
group_by_and_order_by_style = implicit
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
